presto-native-execution/src/test/java/com/facebook/presto/nativeworker/PrestoNativeQueryRunnerUtils.java

```diff
@@ -265,18 +258,51 @@ public static String startRemoteFunctionServer(String remoteFunctionServerBinary
             log.info("Temp directory for Remote Function Server: %s", tempDirectoryPath.toString());
 
             Process p = new ProcessBuilder(Paths.get(remoteFunctionServerBinaryPath).toAbsolutePath().toString(), "--uds_path", remoteFunctionServerUdsPath.toString(), "--function_prefix", REMOTE_FUNCTION_CATALOG_NAME + ".schema.")
-                                .directory(tempDirectoryPath.toFile())
-                                .redirectErrorStream(true)
-                                .redirectOutput(ProcessBuilder.Redirect.to(tempDirectoryPath.resolve("thrift_server.out").toFile()))
-                                .redirectError(ProcessBuilder.Redirect.to(tempDirectoryPath.resolve("thrift_server.err").toFile()))
-                                .start();
+                    .directory(tempDirectoryPath.toFile())
+                    .redirectErrorStream(true)
+                    .redirectOutput(ProcessBuilder.Redirect.to(tempDirectoryPath.resolve("thrift_server.out").toFile()))
+                    .redirectError(ProcessBuilder.Redirect.to(tempDirectoryPath.resolve("thrift_server.err").toFile()))
+                    .start();
             return remoteFunctionServerUdsPath.toString();
         }
         catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    private static NativeWorkerParameters getNativeWorkerParameters()
+    {
+        Path prestoServerPath = Paths.get(Optional.ofNullable(System.getProperty("PRESTO_SERVER"))
+                        .orElse("_build/debug/presto_cpp/main/presto_server"))
+                .toAbsolutePath();
+        Path dataDirectory = Paths.get(Optional.ofNullable(System.getProperty("DATA_DIR"))
+                        .orElse("target/velox_data"))
+                .toAbsolutePath();
+        Optional<Integer> workerCount = Optional.ofNullable(System.getProperty("WORKER_COUNT")).map(Integer::parseInt);
+        assertTrue(Files.exists(prestoServerPath), format("Native worker binary at %s not found. Add -DPRESTO_SERVER=<path/to/presto_server> to your JVM arguments.", prestoServerPath));
+        log.info("Using PRESTO_SERVER binary at %s", prestoServerPath);
+        if (!Files.exists(dataDirectory)) {
+            assertTrue(dataDirectory.toFile().mkdirs());
+        }
+        assertTrue(Files.exists(dataDirectory), format("Data directory at %s is missing. Add -DDATA_DIR=<path/to/data> to your JVM arguments to specify the path", dataDirectory));
+        log.info("using DATA_DIR at %s", dataDirectory);
+        return new NativeWorkerParameters(prestoServerPath, dataDirectory, workerCount);
+    }
+
+    public static class NativeWorkerParameters
+    {
+        public final Path serverBinary;
+        public final Path dataDirectory;
+        public final Optional<Integer> workerCount;
```

Comment:
`workerCount` can be removed from `NativeWorkerParameters`? It has a default value of 4 if none is provided.

Reply:
the logic for retrieving the worker count was duplicated across this file. Keeping it here prevents misuse/duplication